### PR TITLE
gradient-descent

### PR DESCRIPTION
**Pull request by ninichen2911**
change one-dimensional gradient descent, and add two-dimensional gradient descent.

### CODE REVIEW

File: open-machine-learning-jupyter-book/assignments/ml-fundamentals/ml-linear-regression-2.ipynb [hunk 1]
```diff
@@ -5,270 +5,300 @@
    "cell_type": "markdown",
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/ml-fundamentals/ml-linear-regression-2.ipynb:5]: <p>change Assignment to assignment</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/201/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/ml-fundamentals/ml-linear-regression-2.ipynb:5]: <pre class="ql-syntax" spellcheck="false">Line #6.        return 2*x + 5
</pre><p>Please format all your code. https://github.com/open-academy/machine-learning/blob/main/STYLE_GUIDE.md#code</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/201/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/ml-fundamentals/ml-linear-regression-2.ipynb:5]: <p>The step 2 is not the same style as others</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/201/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (reveurmichael) [reply open-machine-learning-jupyter-book/assignments/ml-fundamentals/ml-linear-regression-2.ipynb:5]: <p>In this assignment, we will explain the principle of one-dimensional and two-dimentional gradient descent.</p>

**Review by reveurmichael — APPROVED** [review]

**Pull request merged**